ext/ffi/fast_call.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use std::ffi::c_void;
+use std::iter::once;
+
+use deno_core::v8::fast_api;
+use dynasmrt::dynasm;
+use dynasmrt::DynasmApi;
+use dynasmrt::ExecutableBuffer;
+
+use crate::NativeType;
+use crate::Symbol;
+
+pub(crate) fn is_compatible(sym: &Symbol) -> bool {
+  cfg!(all(target_arch = "x86_64", target_family = "unix"))
+    && !sym.can_callback
+    && !sym.parameter_types.iter().any(|t| !is_fast_api_arg(*t))
+    && is_fast_api_rv(sym.result_type)
+}
+
+pub(crate) fn compile_trampoline(sym: &Symbol) -> Trampoline {
+  Trampoline::compile(sym)
+}
+
+pub(crate) fn make_template(sym: &Symbol, trampoline: &Trampoline) -> Template {
+  let args = once(fast_api::Type::V8Value)
+    .chain(sym.parameter_types.iter().map(|t| t.into()))
+    .collect::<Vec<_>>();
+
+  Template {
+    args: args.into_boxed_slice(),
+    ret: (&fast_api::Type::from(&sym.result_type)).into(),
+    symbol_ptr: trampoline.ptr(),
+  }
+}
+
+/// Trampoline for fast-call FFI functions
+///
+/// Removes first argument (Javascript object receiver) and shifts the rest of arguments to the left
+pub(crate) struct Trampoline(ExecutableBuffer);
+
+impl Trampoline {
+  fn ptr(&self) -> *const c_void {
+    &self.0[0] as *const u8 as *const c_void
```

Comment:
**nitpick:** While rusty_v8 might expect the functions to be given as `*const c_void`, it's not strictly correct. `fn()` is proper. (There may be a difference on some architectures, though not on any that Deno supports.)

Reply:
The type update should be upstreamed?

Reply:
Yes, I think so. Eventually :)

Reply:
I mean, please do open an issue there referencing what's already open in /deno? I'll get around to it when I'm free (unless someone else does first).

Reply:
https://github.com/denoland/rusty_v8/issues/1038